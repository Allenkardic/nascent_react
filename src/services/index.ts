// export const { useGetPokemonByNameQuery } = pokemonApi;
export { default as pokemonApi } from './pokemon';
