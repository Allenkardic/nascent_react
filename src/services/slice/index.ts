export * from './pokemonSlice';
export * from './singlePokemonSlice';
export * from './savedPokemonSlice';
