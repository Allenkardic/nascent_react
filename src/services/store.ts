import { configureStore, Action } from '@reduxjs/toolkit';
import { useDispatch, useSelector } from 'react-redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { ThunkAction } from 'redux-thunk';

import rootReducer, { RootState } from './root-reducer';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['pokemon'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: false,
      immutableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector = () => useSelector;

export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>;
export const persistor = persistStore(store);
