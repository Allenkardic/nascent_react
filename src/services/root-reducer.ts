import { combineReducers } from '@reduxjs/toolkit';

import { pokemonSliceReducer, singlPokemonSliceReducer, savedPokemonSliceReducer } from './slice';

const rootReducer = combineReducers({
  pokemon: pokemonSliceReducer,
  singlePokemon: singlPokemonSliceReducer,
  savedPokemon: savedPokemonSliceReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
