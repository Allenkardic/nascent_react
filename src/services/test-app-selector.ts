import { pokemonSliceReducer, singlPokemonSliceReducer, savedPokemonSliceReducer } from './slice';
const state = {
  pokemon: pokemonSliceReducer,
  singlePokemon: singlPokemonSliceReducer,
  savedPokemon: savedPokemonSliceReducer,
};

export const testAppSelector = (f: any) => f(state);
