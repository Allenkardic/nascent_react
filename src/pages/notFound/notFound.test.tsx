import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { NotFound } from '..';
import { store } from '../../services';
import 'jest-styled-components';

test('it renders NotFound page correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
