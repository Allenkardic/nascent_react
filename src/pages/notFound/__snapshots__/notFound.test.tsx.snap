// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders NotFound page correctly 1`] = `
.c0 {
  padding-top: 16px;
  background-color: #FFFFFF;
  display: flex;
  flex-direction: column;
  margin-left: auto;
  margin-right: auto;
  padding-bottom: 150px;
}

.c3 {
  font-weight: 700;
  color: #333333;
  font-size: 24px;
  text-align: center;
  font-family: Medium;
}

.c4 {
  font-weight: 500;
  color: #4F4F4F;
  font-size: 16px;
  text-align: center;
  font-family: Medium;
}

.c6 {
  font-weight: 700;
  color: #FFFFFF;
  font-size: 16px;
  text-align: center;
  font-family: Medium;
}

.c5 {
  display: flex;
  display: webkit-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #2A3362;
  background-color: #2A3362;
  border: 1px solid #2A3362;
  height: 56px;
  width: 100%;
  border-radius: 4px;
  transition: all 0.3s ease;
  cursor: pointer;
  opacity: 1;
  margin-bottom: 0px;
}

.c1 {
  padding: 24px;
  border: 1px solid #ececec;
  border-radius: 4px;
  margin-top: 24px;
}

.c2 {
  margin-bottom: 16px;
  display: flex;
  display: webkit-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

@media (min-width: 48rem) {
  .c3 {
    font-size: 24px;
  }
}

@media (min-width: 48rem) {
  .c4 {
    font-size: 16px;
  }
}

@media (min-width: 48rem) {
  .c6 {
    font-size: 16px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <svg>
          notFound.svg
        </svg>
      </div>
      <div
        class="c3"
        fw="700"
        ta="center"
      >
        No page found
      </div>
      <div
        class="c4"
        color="#4F4F4F"
        fw="500"
        style="margin-top: 16px; margin-bottom: 4px;"
        ta="center"
      >
        This page could not be found. Go back to the
      </div>
      <div
        class="c4"
        color="#4F4F4F"
        fw="500"
        style="margin-bottom: 32px;"
        ta="center"
      >
        home page
      </div>
      <button
        btnht="56px"
        class="c5"
        type="button"
      >
        <div
          class="c6"
          color="#FFFFFF"
          fw="700"
          ta="center"
        >
          Go back home
        </div>
      </button>
    </div>
  </div>
</div>
`;
